complete/src/test/java/com/example/uploadingfiles/storage/FileSystemStorageServiceTests.java

```diff
@@ -55,10 +55,18 @@ public void saveAndLoad() {
 	}
 
 	@Test
-	public void saveNotPermitted() {
+	public void saveRelativePathNotPermitted() {
 		assertThrows(StorageException.class, () -> {
 			service.store(new MockMultipartFile("foo", "../foo.txt",
-			MediaType.TEXT_PLAIN_VALUE, "Hello, World".getBytes()));
+					MediaType.TEXT_PLAIN_VALUE, "Hello, World".getBytes()));
+		});
+	}
+
+	@Test
+	public void saveAbsolutePathNotPermitted() {
```

Comment:
Since you are highlighting two problems in the related ticket, it might be valuable to have two tests to confirm that both problems are fixed. This can also be valuable from an education standpoint for those reading the unit tests to understand the security principle.

Reply:
Not a problem for adding such a unit test. But I am concerned that this may be a distraction from the main goal of this project.

Reply:
I will fix failed build today. Testing it only on the Windows was not a good idea ;)